### PR TITLE
Added the functions and classes to be used for training and explaining

### PR DESCRIPTION
**Pull request by kebench**
This pull request introduces a reproducible experiment workflow for image classification, centered around a new configuration-driven training script, metrics and evaluation utilities, and reporting tools. The changes enable running, tracking, and analyzing deep learning experiments in a standardized way, with support for class balancing, augmentation, and comprehensive reporting.

The most important changes are:

**Experiment Workflow and Configuration**

* Added a new experiment YAML config (`configs/experiments/exp001_ckplus_resnet18.yaml`) specifying dataset paths, model parameters, training hyperparameters, augmentation, and explainability settings. This enables fully reproducible and configurable experiments.

* Introduced `scripts/train.py`, a command-line script that runs end-to-end training using the config file, handles class balancing (sampler or class weights), saves checkpoints and metrics, and outputs both machine- and human-readable summaries.

**Evaluation and Metrics**

* Added `src/xai_lab/core/engine.py` with reusable functions for training and evaluation (`train_one_epoch`, `evaluate`), returning structured results (loss, accuracy, macro-F1, confusion matrix).

* Added `src/xai_lab/core/metrics.py` with confusion matrix and per-class precision/recall/F1 computation utilities, supporting robust evaluation especially for imbalanced datasets.

**Reporting and Usability**

* Added `scripts/make_report_plots.py`, a standalone script to generate visual plots (loss/accuracy curves, confusion matrix heatmaps) from run outputs, decoupling reporting from training and improving reproducibility and debugging.

* Updated documentation: added troubleshooting tips for explainability artifacts in `artifacts/explainability/exp001_resnet18_saliency_ckplus/README.md` and clarified that scripts add `src/` to `PYTHONPATH` for out-of-the-box usability in `notebooks/README.md`. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R117-R121) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R30-R31)

### CODE REVIEW

File: src/xai_lab/data/transforms/image.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      For facial emotion, over-aggressive crops/warps can remove key cues
+      (eyes/mouth) and make both training and XAI explanations messy/noisy.
+    """
+    crop_scale_min: float = 0.85
+    hflip_p: float = 0.5
+    rotation_deg: int = 10
+    jitter_brightness: float = 0.15
+    jitter_contrast: float = 0.15
+
+
```
Comment (kebench) [inline src/xai_lab/data/transforms/image.py]: Would be good to add why you use these transformation/augmentation. It's for your future self so that he'll understand what goes in here.

File: src/xai_lab/explainers/gradcam.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
```
Comment (kebench) [inline src/xai_lab/explainers/gradcam.py]: What would be the output of GradCAM?

File: README.md [hunk 3]
```diff
@@ -112,6 +112,11 @@ artifacts/explainability/exp001_resnet18_saliency_ckplus/
   ...
 ```
 
+What to do if signals are bad:
```
Comment (kebench) [inline README.md]: signals of what?

File: src/xai_lab/data/transforms/image.py (hunk 1 above)
Comment (kebench) [reply src/xai_lab/data/transforms/image.py]: Added comments for the augmentations.
